avocado/core/loader.py

```diff
@@ -44,6 +44,52 @@
 ALL = True
 
 
+def filter_test_tags(test_suite, filter_by_tags, ignore_empty=False):
+    """
+    Filter the existing (unfiltered) test suite based on tags
+
+    The filtering mechanism is limited to INSTRUMENTED tests, that is it does
+    not apply to SIMPLE tests or any other test type.  It mean non-INSTRUMENTED
+    tests are *always* included.
+
+    :param test_suite: the unfiltered test suite
+    :type test_suite: dict
+    :param filter_by_tags: the list of tag sets to use as filters
+    :type filter_by_tags: list of comma separated tags (['foo,bar', 'fast'])
+    :param ignore_empty: if instrumented tests without tags should not be
```

Comment:
Can I suggest "include empty" as a more intuitive name? "ignore" is ambiguous because it could be "ignore in filtering" (means include) or "ignore in final list" (means exclude).